registrations/tasks.py

```diff
@@ -390,3 +393,164 @@ def run(
+
+    def get_voice_times(self, time):
+        return {'9-11am': '9_11',
+                '2-5pm': '2_5',
+                '6-8pm': '6_8'}.get(time, time)
+
+    def get_voice_days(self, days):
+        return {'monday_and_wednesday': 'mon_wed',
+                'tuesday_and_thursday': 'tue_thu'}.get(days, days)
+
+    def get_receiver(self, receiver):
+        return {
+            "mother_and_father": "mother_father",
+            "mother_and_family": "mother_family",
+            "mother_and_friend": "mother_friend",
+        }.get(receiver, receiver)
+
+    def get_data(self):
+        url = settings.THIRDPARTY_REGISTRATIONS_URL
+        username = settings.THIRDPARTY_REGISTRATIONS_USER
+        password = settings.THIRDPARTY_REGISTRATIONS_PASSWORD
+
+        response = requests.get(url, verify=False, auth=(username, password))
+
+        wb = load_workbook(filename=BytesIO(response.content))
+        ws = wb.get_sheet_by_name('Forms')
+
+        columns = []
+        header_row = ws[1]
+        for cell in header_row:
+            columns.append(cell.value.replace('form.', ''))
+
+        data = []
+
+        for row in ws.iter_rows(min_row=2):
+            reg = {}
+            for i, column in enumerate(columns):
+                reg[column] = row[i].value
+
+            data.append(reg)
+
+        return data
+
+    def run(self, user_id, **kwargs):
+        data = self.get_data()
+
+        source = Source.objects.get(user=user_id)
+
+        loop_error = None
+
+        for line in data:
+
+            try:
+                operator_id = self.get_or_create_identity(
+                    line['health_worker_phone_number'])
+
+                language = self.get_language(line['preferred_msg_language'])
+                receiver = self.get_receiver(line['message_receiver'])
+
+                identity_details = {
+                    "default_addr_type": "msisdn",
+                    "operator_id": operator_id,
+                    "preferred_language": language,
+                    "receiver_role": "mother",
+                    "gravida": line['gravida'],
+                }
+
+                mother_id = self.get_or_create_identity(
+                    line['mothers_phone_number'], identity_details)
+
+                msg_type = self.get_msg_type(line['preferred_msg_type'])
+
+                reg_info = {
+                    "stage": line['type_of_registration'],
+                    "mother_id": mother_id,
+                    "source": source.id,
+                    "data": {
+                        "msg_receiver": receiver,
+                        "operator_id": operator_id,
+                        "language": language,
+                        "msg_type": msg_type
+                    }
+                }
+
+                if line['gatekeeper_phone_number']:
+                    identity_details["receiver_role"] = receiver.replace(
+                        'mother_', '')
```

Comment:
This is only the receiver role of the relevant identity as per:
https://github.com/praekelt/mama-ng-jsbox/blob/develop/go-voice_registration.js#L680
I'm reshuffling some things to update the identity details correctly, commit to follow soon.